api/v1alpha1/groupversion_info.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1alpha1 contains API Schema definitions for the apps v1alpha1 API group
+// +kubebuilder:object:generate=true
+// +groupName=apps.k8s.io
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// GroupVersion is group version used to register these objects
+	GroupVersion = schema.GroupVersion{Group: "apps.k8s.io", Version: "v1alpha1"}
```

Comment:
I'm checking whether it should be apps.k8s.io or app.k8s.io to be consistent with other CRDs.

Reply:
Sure. Quick pointer, the KEP also referred to this as `apiVersion: apps.k8s.io/v1alpha1`.

Reply:
Yeah, apps.k8s.io looks right.  Just want to double check as I found "app.k8s.io" is used for other CRDs.

Reply:
It is "app.k8s.io" here:  https://github.com/kubernetes-sigs/application/blob/master/pkg/apis/app/v1beta1/register.go#L34.

Reply:
is `apps.k8s.io` correct or should we change to `app.k8s.io`?

Reply:
@kubernetes-sigs/application-admins Is there a reason we can't use `apps.k8s.io` and App CRD uses `app.k8s.io`?

Reply:
I think we can go forward with `apps.k8s.io`.
@xing-yang can I mark this as resolved?

Reply:
Would be great to have someone from @kubernetes-sigs/application-admins to confirm this.

Reply:
Don't we already have an `apps` group for builtins? Adding this to `apps.k8s.io` is super confusing. Even worse if other folks are already using `app.k8s.io`. I really recommend a different group name. AFAICT the names in the kep were not intended to be final and the fact that they are written in the KEP doesn't mean they should be used.

Also if you use something not in *.k8s.io or *.kubernetes.io, the API reviewers will leave you alone and apiserver won't balk at installing it. :)

Reply:
Maybe that's why "app.k8s.io" is used for "Application" CRD, not "apps.k8s.io"?  I don't know the reason, hence raising the question.  Since this ExecutionHook repo belongs to sig-apps, I thought it would be better to stay in app.k8s.io for consistency.

Reply:
Yes, perhaps. Note that a SIG can and should have more than one API group, API groups are more fine grained than SIGs.

Are you sure apps is the right SIG for this and not node? This seems more connected to me with pods and runtimes than with replicasets and deployments. The existing `exec` feature is SIG node.

Reply:
Yes, this is under sig-apps: https://github.com/kubernetes/community/blob/master/sigs.yaml#L194.  The KEP was initially done in sig-storage.  @saad-ali suggested that this should be in sig-apps because we need application specific knowledge to use the execution hooks to quiesce and unquiesce an application.  Of course we should get input from sig-node as well.  I believe there was someone from sig-node who reviewed the KEP.

Reply:
Great, just checking. `lifecycle.apps.k8s.io` might be a good group name.

Reply:
sounds good!